client/client.go

```diff
@@ -244,17 +255,26 @@ func (s3Manager S3Manager) GetBucketRegion(ctx context.Context, bucket string, o
 	return manager.GetBucketRegion(ctx, s3Manager.s3Client, bucket, optFns...)
 }
 
-func NewAwsClient(logger hclog.Logger, accounts []Account) Client {
+func NewAwsClient(logger hclog.Logger) Client {
 	return Client{
 		ServicesManager: ServicesManager{
-			services: ServicesAccountRegionMap{},
+			services: ServicesPartitionAccountRegionMap{},
 		},
-		logger:   logger,
-		Accounts: accounts,
+		logger: logger,
+	}
+}
+
+func (s ServicesPartitionAccountRegionMap) Accounts() []string {
+	accounts := make([]string, 0)
+	for partitions := range s {
+		for account := range s[partitions] {
+			accounts = append(accounts, account)
+		}
 	}
+	return accounts
```

Comment:
do we care about duplicates here?

Reply:
I don't think an AWS account can span multiple partitions